### PR TITLE
Add `disableEditing` option to class `TextEditor`

### PR DESCRIPTION
**Pull request by mikeylove**

### CODE REVIEW

File: Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua [hunk 1]
```diff
@@ -294,7 +294,7 @@ function GUI.TextEditor:ontype(char, mod)
 		end
 
 	-- Typeable chars
-	elseif GUI.clamp(32, char, 254) == char then
+	elseif not disableEditing and GUI.clamp(32, char, 254) == char then
```
Comment (ramen) [inline Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua]: This should be self.disableEditing

**Review by ramen — CHANGES_REQUESTED** [review]

File: Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua (hunk 1 above)
Comment (mikeylove) [reply Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua]: good catch, i probably should have copypasted from the vendor version in the other repo

File: Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua [hunk 2]
```diff
@@ -6,7 +6,7 @@
     https://github.com/jalovatt/Lokasenna_GUI/wiki/TextEditor
 
     Creation parameters:
-	name, z, x, y, w, h[, text, caption, pad]
+	name, z, x, y, w, h[, text, caption, pad, disableEditing]
```
Comment (ramen) [inline Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua:9]: This comment implies that disableEditing can be supplied as a positional parameter, but that won't be the case unless you also:

- modify the constructor to take this parameter
- add a line to the constructor to do `txt.disableEditing = txt.disableEditing or disableEditing`

**Review by ramen — CHANGES_REQUESTED** [review]

File: Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua (hunk 2 above)
Comment (mikeylove) [reply Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua:9]: oh, i interpreted that as "anything in the brackets is optional"

Comment (ramen) [reply Lokasenna_GUI v2/Library/Classes/Class - TextEditor.lua:9]: There are some parameters that are "Optional", and then more under "Additional parameters":

https://github.com/jalovatt/Lokasenna_GUI/wiki/2.00-TextEditor#additional-parameters

Both sets are optional, but only the ones in the constructor parameter list can be passed as positional parameters.

**Review by ramen — APPROVED** [review]

**Pull request merged**